tensorbay/opendataset/Cityscapes/loader.py

```diff
@@ -105,3 +107,92 @@ def CityscapesGTFine(path: str) -> Dataset:
 
             segment.append(data)
     return dataset
+
+
+def CityscapesGTCoarse(path: str) -> Dataset:
+    """`CityscapesGTCoarse <https://www.cityscapes-dataset.com/>`_ dataset.
+
+    The file structure should be like::
+
+        <path>
+            leftImg8bit/
+                train/
+                    aachen/
+                        aachen_000000_000019_leftImg8bit.png
+                        ...
+                    ...
+                train_extra/
+                    augsburg/
+                        augsburg_000000_000019_leftImg8bit.png
+                        ...
+                    ...
+                val/
+                    frankfurt/
+                        frankfurt_000000_000019_leftImg8bit.png
+                        ...
+                    ...
+                ...
+            gtCoarse/
+                train/
+                    aachen/
+                        aachen_000000_000019_gtCoarse_instanceIds.png
+                        aachen_000000_000019_gtCoarse_labelIds.png
+                        aachen_000000_000019_gtCoarse_polygons.json
+                        ...
+                    ...
+                train_extra/
+                    augsburg/
+                        augsburg_000000_000019_gtCoarse_instanceIds.png
+                        augsburg_000000_000019_gtCoarse_labelIds.png
+                        augsburg_000000_000019_gtCoarse_polygons.json
+                        ...
+                    ...
+                val/
+                    frankfurt/
+                        frankfurt_000000_000019_gtCoarse_instanceIds.png
+                        frankfurt_000000_000019_gtCoarse_labelIds.png
+                        frankfurt_000000_000019_gtCoarse_polygons.json
+                        ...
+                    ...
+            ...
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.join(os.path.abspath(os.path.expanduser(path)))
+
+    dataset = Dataset(DATASET_NAME_GTCoarse)
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+
+    for segment_name in _SEGMENT_NAMES_GTCoarse:
+        segment = dataset.create_segment(segment_name)
+        for image_path in glob(os.path.join(root_path, "leftImg8bit", segment_name, "*", "*.png")):
+            city = os.path.basename(image_path).split("_", 1)[0]
```

Comment:
`os.path.basename(image_path)` is calculated twice, can we assign it to a variable?